tests/unit/components/ParameterForm.spec.ts

```diff
@@ -147,12 +132,46 @@ describe("parameter form", () => {
     await countrySelect.findAll("option").at(0)!.setSelected();
     await component.findComponent({ name: "CButtonGroup" }).find("input[value='yes']").setChecked();
 
-    formData = JSON.parse(cForm.element.attributes.getNamedItem("data-test")!.value);
+    formData = JSON.parse(cForm.element.attributes.getNamedItem("data-test-form-data")!.value);
     expect(formData.region).toBe("CLD");
     expect(formData.long_list).toBe("3");
     expect(formData.short_list).toBe("yes");
   });
 
+  it("sends a POST request to /api/scenarios with the form data when submitted", async () => {
+    registerEndpoint("/api/scenarios", {
+      method: "POST",
+      handler: (event) => {
+        const query = getQuery(event) as Record<string, string>;
+        const modelParams = JSON.parse(query.parameters);
+
+        if (modelParams.long_list === "1" && modelParams.region === "HVN" && modelParams.short_list === "no") {
+          return { runId: "randomId" };
+        } else {
+          return { error: "Test failed due to wrong parameters" };
+        }
+      },
+    });
+
+    const component = await mountSuspended(ParameterForm, {
+      props: { metadata, metadataFetchStatus: "success", metadataFetchError: null },
+      global: { stubs },
+    });
+
+    const buttonEl = component.find("button[type='submit']");
+    expect(buttonEl.attributes("disabled")).not.toBe("");
+    await buttonEl.trigger("click");
+    expect(buttonEl.attributes("disabled")).toBe("");
+
+    await flushPromises();
+    // I couldn't find a way to spy on the navigateTo function, so this is a second-best test that we
+    // at least construct the correct path to navigate to.
```

Comment:
Did you try stubGlobal or mockNuxtImport?
https://stackoverflow.com/questions/77827093/how-to-check-if-navigateto-was-called-in-nuxt-js

Reply:
Yes.

Reply:
OK, let's not hold up this branch, but it would be really good to find a way to mock global methods, and it seems really weird if the alleged mocking capacities of vitest/nuxt test utils aren't going to let us to that. Tbh if that proves impossible personally I'd rather not clutter up the code with extra props for the sake of unit tests, as long as the navigation etc gets tested in e2e tests. These components are going to be busy and confusing enough as it is, without a bunch of extra stuff that doesn't really need to be there.